### PR TITLE
Added support for DVD images export as DVD-ROM drives

### PR DESCRIPTION
**Pull request by mr-const**
Implementation of several commands which provide access to DVD devices using iSCSI protocol.
Added optical disc image entity to allow simple export of DVD images.

### CODE REVIEW

**Comment by TalAloni** [comment]
Nice! I'll take a look during the weekend

File: ISCSI/SCSI/SCSICommandDescriptorBlock/SCSICommandDescriptorBlock.cs [hunk 1]
```diff
@@ -73,8 +75,14 @@ public static SCSICommandDescriptorBlock FromBytes(byte[] buffer, int offset)
...
                     return new SCSICommandDescriptorBlock12(buffer, offset);
+                case SCSIOpCodeName.ReadDiscInformation:
+                    return new SCSICommandDescriptorBlock12(buffer, offset);
+                case SCSIOpCodeName.ReadToc:
+                    return new ReadTocCommand(buffer, offset);
+                case SCSIOpCodeName.GetConfiguration:
+                    return new GetConfigurationCommand(buffer, offset);
                 default:
-                    throw new UnsupportedSCSICommandException(String.Format("Unknown SCSI command: 0x{0}", opCode.ToString("x")));
+                    throw new UnsupportedSCSICommandException($"Unknown SCSI command: 0x{opCode:X}");
```
Comment (TalAloni) [inline ISCSI/SCSI/SCSICommandDescriptorBlock/SCSICommandDescriptorBlock.cs:85]: If it ain't broke don't try to fix it please. while C# 2.0 may seem obsolete at this point, the entire project is compatible with it so I'm not sure I want to break it.

File: ISCSI/SCSI/SCSIReturnParameters/StandardInquiryData.cs [hunk 2]
```diff
@@ -119,7 +119,7 @@ public byte[] GetBytes()
         {
             byte[] buffer = new byte[96];
             buffer[0] |= (byte)(PeripheralQualifier << 5);
-            buffer[0] |= (byte)(PeripheralQualifier & 0x1F);
+            buffer[0] |= (byte)((int)PeripheralDeviceType & 0x1F);
```
Comment (TalAloni) [inline ISCSI/SCSI/SCSIReturnParameters/StandardInquiryData.cs:122]: This is a bugfix so it warrants a separate PR - I went ahead and pushed the fix to save you the trouble.

**Comment by TalAloni** [comment]
I find it hard to review large PRs - anything that can be added separately, like bugfixes, addition of enums and etc. would be best pushed in a separate PR

**Comment by TalAloni** [comment]
Also, do you have any copyright claims to the changes above, or are you providing them as public domain?

File: ISCSI/SCSI/SCSICommandDescriptorBlock/SCSICommandDescriptorBlock.cs (hunk 1 above)
Comment (mr-const) [reply ISCSI/SCSI/SCSICommandDescriptorBlock/SCSICommandDescriptorBlock.cs:85]: Sure, I'll revert this change

**Comment by mr-const** [comment]
I don't have any copyright claims. Regarding separate PRs I agree, but on the other hand this DVD functionality is pretty hard to decompose, I'll take a look anyway.

**Comment by TalAloni** [comment]
also - and this should be tackled in a separate PR for the UI - I would have probably created an "Add Virtual CD/DVD" rather than use the "Add Virtual Disk" button. (and the "Read Only" checkbox is irrelevant for CD/DVD - it shouldn't be active there)

**Pull request closed without merging**